extensions/positron-notebook-controllers/src/utils.ts

```diff
@@ -13,3 +16,27 @@ export function formatCount(count: number, unit: string): string {
 	}
 	return `${count} ${unit}s`;
 }
+
+/**
+ * Get the language runtime session for a notebook.
+ *
+ * @param notebookUri The URI of the notebook.
+ * @param runtimeId Optional runtime ID to filter the session by.
+ * @returns Promise that resolves with the language runtime session, or `undefined` if no session is found.
+ */
+export async function getNotebookSession(
+	notebookUri: vscode.Uri, runtimeId?: string,
+): Promise<positron.LanguageRuntimeSession | undefined> {
+	// Get the session for the notebook.
+	const session = await positron.runtime.getNotebookSession(notebookUri);
+	if (!session) {
+		return undefined;
+	}
+
+	// Ensure that the session is for the requested runtime.
+	if (runtimeId && session.runtimeMetadata.runtimeId !== runtimeId) {
+		return undefined;
```

Comment:
Is this a case worthy of logging some sort of message for?

Reply:
Good idea